packages/react-ui/src/widgets/BarWidgetUI.js

```diff
@@ -0,0 +1,418 @@
+          )}
+        </Grid>
+      )}
+      {!!options && (
+        <ReactEcharts
+          option={options}
+          lazyUpdate={true}
+          onEvents={onEvents}
+          style={{ height }}
+        />
+      )}
+    </div>
+  );
+}
+
+BarWidgetUI.defaultProps = {
+  tooltip: true,
+  tooltipFormatter: defaultTooltipFormatter,
+  yAxisData: [],
+  series: [],
+  xAxisFormatter: (v) => v,
+  yAxisFormatter: (v) => v,
+  selectedBars: [],
+  labels: {},
+  onSelectedBarsChange: null,
+  animation: true,
+  vertical: true,
+  stacked: true
+};
+
+BarWidgetUI.propTypes = {
+  yAxisData: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.number)),
+    PropTypes.arrayOf(PropTypes.number)
+  ]).isRequired,
+  xAxisData: PropTypes.arrayOf(PropTypes.string).isRequired,
+  series: PropTypes.arrayOf(PropTypes.string),
+  colors: PropTypes.arrayOf(PropTypes.string),
+  stacked: PropTypes.bool,
+  vertical: PropTypes.bool,
+  labels: PropTypes.object,
+  tooltip: PropTypes.bool,
+  tooltipFormatter: PropTypes.func,
+  xAxisFormatter: PropTypes.func,
+  yAxisFormatter: PropTypes.func,
+  selectedBars: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.number)),
+    PropTypes.arrayOf(PropTypes.number)
+  ]),
+  onSelectedBarsChange: PropTypes.func,
+  height: PropTypes.number,
+  animation: PropTypes.bool
+};
+
+export default BarWidgetUI;
+
+// Aux
+
+function useProcessedProps(props) {
+  const theme = useTheme();
+  const {
+    labels,
+    height,
+    selectedBars: _selectedBars,
+    yAxisData: _yAxisData,
+    colors: _colors,
+    series: _series
+  } = props;
+
+  // Format series with labels, we don't need original series labels in the widget
+  const series = useMemo(() => _series.map((name) => labels[name] || name), [
+    _series,
+    labels
+  ]);
+
+  // Use yAxisData always as a two-dimensions array
+  const yAxisData = useMemo(
+    () => (Array.isArray(_yAxisData[0]) ? _yAxisData : [_yAxisData]),
+    [_yAxisData]
+  );
+
+  // Colors
+  const colors = useMemo(() => {
+    // Use colors props if exists
+    if (_colors) return _colors;
+
+    // Use secondary.main, if yAxisData is flat or series has a unique element
+    if (yAxisData.length <= 1 || (series.lenght && series.length === 1))
+      return [theme.palette.secondary.main];
+
+    // Use secondary light and dark, if yAxisData is bidimensional or series has two elements
+    if (yAxisData.length === 2 || (series.length && series.length === 2))
+      return [theme.palette.secondary.light, theme.palette.secondary.dark];
+
+    // Use qualitate palette with mutidimensional bar chart
+    return Object.values(theme.palette.qualitative.bold || {});
+  }, [_colors, theme, yAxisData, series]);
+
+  return {
+    ...props,
+    height: height ?? theme.spacing(22),
+    selectedBars: formatSelectedBars(_selectedBars),
+    yAxisData,
+    colors,
+    series
+  };
+}
+
+function formatSelectedBars(selectedBars) {
```

Comment:
Do we really need this sanitizing stuff?
Why isn't it the responsibility of the calling code to send props in the correct format?

Reply:
To simplify stuff for users.